### DISCUSSION TITLE
Google Auth, Vercel hosting

### DISCUSSION THREAD

**Discussion by OlegPuzankin** [nextauthjs/next-auth#1580] [Help] [answered]
Hello folks

Could you please give some advice on how to handle such a situation?

I create a small next app and made production deployment at this URL - https://comtel-inventory.vercel.app.

Inside of app I have to use Google authentication of users with the Next-Auth library. On the google console, I provide such authorized redirect URI 
https://comtel-inventory.vercel.app/api/auth/callback/google

but when I try to login I see 400 error on google OAuth consent screen: **redirect_uri_mismatch - The redirect URI in the request, https://comtel-inventory-6ohmho13d-olegpuzankin.vercel.app/api/auth/callback/google**

this is one of my deployment URLs - **comtel-inventory-6ohmho13d-olegpuzankin.vercel.app**
Is there some way to use  regular production URL of the web-site in auth process -  https://comtel-inventory.vercel.app, not this strange URL - https://comtel-inventory-6ohmho13d-olegpuzankin.vercel.app

Thank you.

**Comment by StefanSelfTaught** [comment]
Did you set your `NEXTAUTH_URL` environment variable on vercel?

**Reply by OlegPuzankin** [reply]
Yes, I set NEXTAUTH_URL as https://comtel-inventory.vercel.app

**Comment by StefanSelfTaught** [comment, marked as answer]
This is how the `baseUrl` is gets his value. What you have there is the `VERCEL_URL`
```js
const { basePath, baseUrl } = parseUrl(process.env.NEXTAUTH_URL || process.env.VERCEL_URL)
```
Make sure you use that env variable on the production build

**Reply by OlegPuzankin** [reply]
Stefan,

thank you very much for helping me. Did I understand you correct that my production build does not get
 `NEXTAUTH_URL = https://comtel-inventory.vercel.app`
and use `VERCEL_URL=https://comtel-inventory-6ohmho13d-olegpuzankin.vercel.app` instead and this cause that goole auth process point to incorrect redirect URL like
`https://comtel-inventory-6ohmho13d-olegpuzankin.vercel.app/api/auth/callback/google`

How I can see `console.log` on running production build to check if NEXTAUTH_URL  has the correct value?

**Reply by OlegPuzankin** [reply]
It seems like I find what's a problem, after I added the NEXTAUTH_URL variable I did not do a new production deployment.
Now it seems to me redirect work OK.
What I can not understand why after success login and redirect to the home page it is added hash (#) symbol to URL in browser 
like 
`https://comtel-inventory.vercel.app/#`

**Reply by StefanSelfTaught** [reply]
> It seems like I find what's a problem, after I added the NEXTAUTH_URL variable I did not do a new production deployment.

Yep that was what I was saying

I don't really know about the hash, you can google it, but it is a problem?

**Reply by OlegPuzankin** [reply]
Hash symbol is not a problem at all). Thanks for the help!
